### ISSUE TITLE
Duplicating Documents which have been wrapped with additional methods...

### ISSUE THREAD

**Issue by DanielDornhardt** [msavin/Mongol#16]
seems to lead to an error.

The issue seems to be that the "duplicate" - code runs through all the objects' properties as returned by Collection.find() . But unfortunately I wrap those with a class and additional methods using meteors' "transform" - function.

As I don't see the functions and additional data fields added by the constructor in the json for updating the documents via mongol I think that there might be another way / maybe a way to skip the transform-function?

Best wishes

Daniel

**Comment by wallslide** [comment]
I also ran into a problem with transformed objects. I have a parent object that has an array of sub-objects, and I store a reference to the parent object in the sub objects. Unfortunately I get a circular reference error which doesn't allow me to use that collection with Mongol. If Mongol would either display collection documents pre-transformation, or give me a hook to de-transform the object before Mongol attempts to display it then that would make it a lot more useful for me.

**Comment by msavin** [comment]
I'll look into this when I have bit more time. Thanks for letting me know!

**Comment by JackAdams** [comment]
@DanielDornhardt did that error message have EJSON somewhere in it?  My guess is the we're trying to send functions (those extra methods you added) from the client to the server via DDP and, of course, functions aren't a valid EJON type.

The reason they're not displaying in the json for updating documents is that the document gets thrown into `JSON.stringify(docCurrent, null, 2);` which just strips out all the properties that have function values. (And then the resulting JSON is displayed using a `<pre> ... </pre>` codeblock.)

Anyway, long story short, if we strip out the functions before sending the document back to the server via DDP, everything should work fine.

I'll add the fix to a PR I've got open.
